src/main/java/play/dev/filewatch/PollingFileWatchService.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.dev.filewatch;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/** A polling Play watch service. Polls in the background. */
+public class PollingFileWatchService implements FileWatchService {
+  private final int pollDelayMillis;
+
+  private volatile boolean stopped;
+
+  public PollingFileWatchService(int pollDelayMillis) {
+    this.pollDelayMillis = pollDelayMillis;
+  }
+
+  private static Iterable<File> listRecursively(Iterable<File> files) {
```

Comment:
I would not call the argument `sourcesFinder` because it's not a Supplier (see my other comment)